### ISSUE TITLE
Dev branch: Error: no such table: main.group__ on `pihole -g`

### ISSUE THREAD

**Issue by timocapa** [pi-hole/pi-hole#3059]
First of all, I apologize if this is on the wrong repo.

**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

`1`

---
**Expected behaviour:**

`pihole -g runs fine and finishes updating lists`

**Actual behaviour:**

`  [i] Neutrino emissions detected...
  [✗] Pulling blocklist source list into range
  [i] No source list found, or it is empty

Error: no such table: main.group__
Error: no such table: main.group__
  [i] Number of gravity domains:  ( unique domains)
Error: no such table: main.group__
  [i] Number of exact blacklisted domains:
Error: no such table: main.group__
  [i] Number of regex blacklist filters:
Error: no such table: main.group__
  [i] Number of exact whitelisted domains:
Error: no such table: main.group__
  [i] Number of regex whitelist filters:
  [✓] Cleaning up stray matter

  [✓] Flushing DNS cache
  [✗] DNS service is NOT running`

**Steps to reproduce:**

`Update to latest dev commits (at which point it should fail gravity already)`

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

`8rsz9gitpg`

**Troubleshooting undertaken, and/or other relevant information:**

`pihole -r and reconfiguring, reverting to master makes it work again`

**Additional info:**
I can reproduce this on both my OrangePi Zero, and my "redundant" VM in FreeNAS. Both run Debian 10 converted to DietPi, **however** I used the original install script of PiHole, and not from dietpi-software.

**Comment by PromoFaux** [comment]
Thanks for the report, we've actually worked out the cause to this one already and are working on a fix.. watch this space :)

**Comment by timocapa** [comment]
Hey, wow, that was a fast reply. Thanks! 

I try to keep one of my 2 piholes on dev (preferably both) to try to catch bugs quickly to try to help out :).

**Comment by PromoFaux** [comment]
Heh, the beauty of having new issue notifications pop up in our chatops server.

The issue is an unintended side-effect of a temporary rename and drop of one of the tables. Turns out that sqlite3 likes to be clever and update any references to a renamed table elsewhere in the database. Hopefully we can fix it and preserve data.

**Comment by PromoFaux** [comment]
@timocapa, keep an eye on #3060. Once it is approved and merged, it should resolve the issue for you.

**Comment by timocapa** [comment]
I'll go ahead and pick it and see how it goes.

**Comment by timocapa** [comment]
Now I can't say if this is the "official" way you all do it, but I just edited /etc/.pihole/advanced/Scripts/piholeCheckout.sh to say `propsed_8_to_9` instead of development and that seemed to have worked :)

![As far as I can see, it works fine now](https://user-images.githubusercontent.com/22844613/71564621-2f03c200-2aa4-11ea-9cc2-0025f3df82af.png)

**Comment by technicalpyro** [comment]
the uncut method would have been to run the command `pihole checkout core propsed_8_to_9` 

ALSO once you see it merged into the dev branch be sure to rerun `pihole checkout dev`

**Comment by timocapa** [comment]
Alright, good to know for the future, perhaps my way wasn't as good looking but it still worked :)

Sure, thanks for the heads up.

**Comment by timocapa** [comment]
Hmm, seems switching from 8 to 9 back to master and then back to 8 to 9 breaks it again on my VM. Initial upgrade on my OrangePi went fine and I don't really want to touch it as that's my "main" PiHole.
![image](https://user-images.githubusercontent.com/22844613/71564756-3deb7400-2aa6-11ea-8d08-0d27940917bf.png)

**Comment by technicalpyro** [comment]
so the appropriate route would be `master`>`Dev`>`8 to 9` because of how this is coded the transitions have to be made one step at a time

**Comment by timocapa** [comment]
I've only been jumping between 8_to_9 and now neither Dev nor 8_to_9 work after going 8_to_9 once. Only master. Huh. Same case on both the VM and Pi :thinking: 

For my NAS I'm not sure what I did.

For my Pi, I initially upgraded to `8 to 9`, which worked. After I added a blocklist to the GUI (as I lost my settings earlier), PiHole updated gravity again, which immediately failed with the same errors as we initially had.

I'm gonna set up a second VM and snapshot it before I check out to any other branch.

**Comment by timocapa** [comment]
Here's a GIF. This is a full clean install. `Master` -> `Dev` -> `8 to 9` -> `pihole -g`. 

Debian 10 x86_64, no conversion.

![Peek 2019-12-30 02-44](https://user-images.githubusercontent.com/22844613/71565360-5069ab80-2aae-11ea-9023-ece68b717cfa.gif)

**Comment by DL6ER** [comment]
I see #3060 does not update the internal database version. Hence, the script will be run on every `pihole -g` run which does not work. I will add the missing `UPDATE` line there and everything should be solved.

You'd need to run
```sql
sudo sqlite3 /etc/pihole/gravity.db "UPDATE info SET value = 9 WHERE property = 'version';"
```
to get your database into the correct shape.

**Comment by timocapa** [comment]
Seems on my first VM this doesn't seem to work.

![image](https://user-images.githubusercontent.com/22844613/71579936-eb8a7180-2afe-11ea-8ea9-9e07981d4820.png)

**Comment by timocapa** [comment]
I can however confirm that first upgrades work fine.

**Comment by DL6ER** [comment]
On your first VM, the gravity database is now incomplete because no group called `group` exists. What is the output of
```
sqlite3 /etc/pihole/gravity.db ".schema group__"
```
?

**Comment by timocapa** [comment]
Afraid I can't answer this anymore as I removed `gravity.db` and let PiHole recreate it. It's working now.

If it's still any help, `sqlite3 /etc/pihole/gravity.db ".schema group__"` returns empty on the new db.

**Comment by timocapa** [comment]
As I haven't had issues since I think I can close this issue now. :) 

Thanks a lot, guys.
